### PR TITLE
Extract extensions to another package

### PR DESCRIPTION
**Pull request by khenderick**

### CODE REVIEW

File: ovs/extensions/generic/disk.py [hunk 1]
```diff
@@ -16,172 +16,15 @@
...
-        :return: None
-        """
-        try:
-            check_output("umount '{0}'".format(mountpoint.replace(r"'", r"'\''")), shell=True)
-        except Exception:
-            DiskTools._logger.exception('Unable to umount mountpoint {0}'.format(mountpoint))
+    
+    @classmethod
+    def _get_os_manager(cls):
+        return OSFactory.get_manager()
```
Comment (JeffreyDevloo) [inline ovs/extensions/generic/disk.py]: Newline :)

File: ovs/extensions/generic/system.py [hunk 2]
```diff
@@ -39,151 +38,49 @@ def __init__(self):
...
-    @staticmethod
-    def get_my_machine_id(client=None):
+    @classmethod
+    def get_my_machine_id(cls, client=None):
         """
-        Returns unique machine id, generated at install time.
-        :param client: Remote client on which to retrieve the machine ID
+        Returns unique machine id, generated during the setup of the iSCSI Manager
+        :param client: Local client on which to retrieve the machine ID
         :type client: SSHClient
```
Comment (JeffreyDevloo) [inline ovs/extensions/generic/system.py]: @kvanhijf did some copy pasting here?

File: ovs/lib/helpers/toolbox.py [hunk 3]
```diff
@@ -213,22 +212,23 @@ def change_service_state(client, name, state, logger=None):
...
         """
+        service_manager = ServiceFactory.get_manager()
         action = None
-        status = ServiceManager.get_service_status(name, client=client)
+        status = service_manager.get_service_status(name, client=client)
         if status != 'active' and state in ['start', 'restart']:
             if logger is not None:
                 logger.debug('{0}: Starting service {1}'.format(client.ip, name))
-            ServiceManager.start_service(name, client=client)
+                service_manager.start_service(name, client=client)
```
Comment (JeffreyDevloo) [inline ovs/lib/helpers/toolbox.py]: Copy pasted indent? :)
Good thing we have a logger here!

**Review by JeffreyDevloo — CHANGES_REQUESTED** [review]

**Review by JeffreyDevloo — APPROVED** [review]

**Comment by khenderick** [comment]
After reviewing with @kvanhijf and @JeffreyDevloo, this is merged to `develop`. Further tests will be done on that branch.

**Pull request merged**